src/Razor/test/Microsoft.AspNetCore.Razor.Test.Common/Language/IntegrationTests/IntegrationTestBase.cs

```diff
@@ -519,6 +519,70 @@ protected void AssertSourceMappingsMatchBaseline(RazorCodeDocument codeDocument)
             }
         }
 
+        protected void AssertLinePragmas(RazorCodeDocument codeDocument, bool designTime)
+        {
+            if (FileName == null)
+            {
+                var message = $"{nameof(AssertSourceMappingsMatchBaseline)} should only be called from an integration test. ({nameof(FileName)} is null).";
+                throw new InvalidOperationException(message);
+            }
+
+            var csharpDocument = codeDocument.GetCSharpDocument();
+            Assert.NotNull(csharpDocument);
+            var linePragmas = csharpDocument.LinePragmas;
+            designTime = false;
+            if (designTime)
```

Comment:
I am thinking of removing this entire block and using classified spans as the source of truth for design time tests as well. Thoughts?

Reply:
Is the idea that we'd use the spans instead of the source mappings?

What I'd want to know is "do we validate the spans and source mappings together"?

Reply:
> Is the idea that we'd use the spans instead of the source mappings?

Yes

> What I'd want to know is "do we validate the spans and source mappings together"?

Also yes. So my plan is the following,
```
foreach code span:
    verify if a line pragma exists
    if DesignTime:
        verify if a corresponding source mapping exists
```

Reply:
Nevermind. This won't work because there isn't always a 1:1 mapping between SourceMappings and LinePragmas. I am going to stick to what's here already.